### PR TITLE
Updating ACK runtime to v0.14.1

### PR DESCRIPTION
**Pull request by vijtrip2**
Issue #, if available: https://github.com/aws-controllers-k8s/community/issues/982

Description of changes:
* Updating ACK runtime to v0.14.1
* Adding e2e tests for activity and statemachine resources

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: helm/Chart.yaml [hunk 1]
```diff
@@ -1,15 +1,17 @@
 apiVersion: v1
-name: ack-sfn-controller
-description: A Helm chart for the ACK service controller for sfn
-version: v0.0.2
-appVersion: v0.0.2
-home: https://github.com/aws/aws-controllers-k8s
+name: sfn-chart
+description: A Helm chart for the ACK service controller for AWS Step Functions (Step-Functions)
+version: v0.0.0-non-release-version
+appVersion: v0.0.0-non-release-version
```
Comment (jaypipes) [inline helm/Chart.yaml:5]: Hrmmm. Either you didn't do `git fetch --all --tags` or there is no Git tag for v0.0.2 (more worrying)...

File: test/e2e/tests/test_state_machine.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+            exists = state_machine_exists(sfn_client, state_machine)
+            assert exists
+        except:
+            if state_machine is not None:
+                delete_state_machine(state_machine)
+            return pytest.fail("StateMachine failed to create")
+
+        delete_state_machine(state_machine)
+        exists = state_machine_exists(sfn_client, state_machine)
+        assert not exists
```
Comment (jaypipes) [inline test/e2e/tests/test_state_machine.py:112]: nice work on these tests, @vijtrip2 :)

**Review by jaypipes — CHANGES_REQUESTED** [review]
Only concern is the `v0.0.0-non-release` version (see inline comment)

File: test/e2e/bootstrap.yaml [hunk 3]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (RedbackThomson) [inline test/e2e/bootstrap.yaml:1]: This file should not be included in the git. Please do a `git rm`.

File: test/e2e/service_bootstrap.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+#	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+"""Bootstraps the resources required to run the S3 integration tests.
```
Comment (RedbackThomson) [inline test/e2e/service_bootstrap.py:13]:
```suggestion
"""Bootstraps the resources required to run the SFN integration tests.
```

File: test/e2e/service_cleanup.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+#	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+"""Cleans up the resources created by the S3 bootstrapping process.
```
Comment (RedbackThomson) [inline test/e2e/service_cleanup.py:13]:
```suggestion
"""Cleans up the resources created by the SFN bootstrapping process.
```

File: test/e2e/tests/test_activity.py [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+        try:
+            activity = create_activity("activity")
+            assert k8s.get_resource_exists(activity.ref)
+
+            exists = activity_exists(sfn_client, activity)
+            assert exists
+        except:
+            if activity is not None:
+                delete_activity(activity)
+            return pytest.fail("Activity failed to create")
```
Comment (RedbackThomson) [inline test/e2e/tests/test_activity.py:108]: This `try/except` block can be made into a fixture, so we can use a `yield` for catching.
See https://github.com/aws-controllers-k8s/s3-controller/blob/main/test/e2e/tests/test_bucket.py#L104-L105

**Review by RedbackThomson — CHANGES_REQUESTED** [review]
Left some inline comments regarding the tests. Otherwise logic looks clean. This seems to be a fairly straightforward API, actually.

File: helm/Chart.yaml (hunk 1 above)
Comment (vijtrip2) [reply helm/Chart.yaml:5]: There are no tags on this repo. https://github.com/aws-controllers-k8s/sfn-controller/tags
I will use env variable to set this to a released version.

**Review by RedbackThomson — APPROVED** [review]
/lgtm

**Pull request merged**